geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/QueryFilterSplitter.scala

```diff
@@ -223,7 +224,45 @@ class QueryFilterSplitter(sft: SimpleFeatureType) extends Logging {
       }
     }
 
-    makeDisjoint(mergeOverlappedFilters(combineSecondaryFilters(reduceChildOptions(getChildOptions))))
+    // detect the pattern it the pattern (attr in (1,2,3,4,5,6, etc) AND something else)
+    def detectSingleAttrOr: Boolean = {
+      // each child is expressed as (attr = x(i) AND expr2 AND expr3 ...)
+      val attrs =
+        filter.getChildren.map {
+          case and: And => and.getChildren.flatMap {
```

Comment:
We are...but the ```logicDistributionDNF``` method transforms it out into something like this:
```
[
  [
     [ high = h0 ]
     AND [ geom bbox POLYGON ((40 40, 40 50, 50 50, 50 40, 40 40)) ] 
     AND org.geotools.filter.temporal.DuringImpl@e5a72d93
  ] 
  OR 
  [
    [ high = h1 ]
    AND [ geom bbox POLYGON ((40 40, 40 50, 50 50, 50 40, 40 40)) ]
    AND org.geotools.filter.temporal.DuringImpl@e5a72d93
  ]
  OR 
  [
    [ high = h2 ]
```

So really the  match statement is referring to the "children" of the OR being an AND: 
```
[[ high = h0 ] AND [ geom bbox POLYGON ((40 40, 40 50, 50 50, 50 40, 40 40)) ] AND org.geotools.filter.temporal.DuringImpl@e5a72d93]
```

That being said, you bring up a good point that we should verify it is a top level OR filter whose children are  AND predicates.